### PR TITLE
[PHI]Remove OneDNN code in Transpose infershape

### PR DESCRIPTION
**Pull request by YuanRisheng**
### PR types

Others
### PR changes

Others
### Describe

Pcard-67876
Move the OneDNN's code from Infershape To Kernel and eliminate differences between dynamic graph and static graph of transpose's infershape

### CODE REVIEW

**Review by heavyrain-lzy — APPROVED** [review]

**Comment by heavyrain-lzy** [comment]
> Contributor

PR描述里应该加上iCafe Card,不然`CheckPRTemplateRequirements`过不了

**Comment by YuanRisheng** [comment]
> > Contributor
> 
> PR描述里应该加上iCafe Card,不然`CheckPRTemplateRequirements`过不了

好的，谢谢

**Review by zyfncg — APPROVED** [review]

**Comment by zyfncg** [comment]
image
这个是不得用 Pcard ?

**Comment by YuanRisheng** [comment]
> image
> 
> 这个是不得用 Pcard ?

好的，我修改一下

File: paddle/fluid/operators/transpose_op.cc [hunk 1]
```diff
@@ -218,7 +143,12 @@ class Transpose2Op : public TransposeOp {
       : TransposeOp(type, inputs, outputs, attrs) {}
 
   void InferShape(framework::InferShapeContext *ctx) const override {
-    TransposeOp::InferShape(ctx);
+    using CompatMetaTensor = framework::CompatMetaTensor;
```
Comment (chenwhql) [inline paddle/fluid/operators/transpose_op.cc:146]: 这个是不是也可以移除了，XShape在phi里没有了，这个var实际上相当于废了

**Review by chenwhql — APPROVED** [review]

File: paddle/fluid/operators/transpose_op.cc (hunk 1 above)
Comment (YuanRisheng) [reply paddle/fluid/operators/transpose_op.cc:146]: 之前尝试移除发现移不动，于是保留了
image

**Comment by YuanRisheng** [comment]
I merge this PR inadvertently and please help review this PR @Silv3S

File: paddle/phi/infermeta/unary.cc [hunk 2]
```diff
@@ -4228,7 +4228,9 @@ void TransposeInferMeta(const MetaTensor& x,
   int x_rank = x_dims.size();
   int axis_size = axis.size();
 
-  PADDLE_ENFORCE_EQ(
+  // Note: x_rank > axis_size when fuse squeeze2 + transpose2, else x_rank ==
+  // axis_size
```
Comment (Silv3S) [inline paddle/phi/infermeta/unary.cc:4232]: This note doesn't match `PADDLE_ENFORCE` error message below

Comment (YuanRisheng) [reply paddle/phi/infermeta/unary.cc:4232]: Thank you! I will update message in next PR.

**Review by Silv3S — COMMENTED** [review]
LGTM

**Pull request merged**